src/components/Card/Card.tsx

```diff
@@ -1,18 +1,24 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Link } from 'react-router-dom';
 import { Typography } from '@mui/material';
 import classNames from 'classnames';
 import PlayCircleIcon from '@mui/icons-material/PlayCircle';
 import { TypeMovieCard } from '../types';
 import style from './Card.module.scss';
-import { useAppSelector } from '../../store/store';
+import { useAppDispatch, useAppSelector } from '../../store/store';
+import { fetchGenresData } from '../../store/reducers/CardsActions';
 
 function Card(movieData: TypeMovieCard) {
   const { title, vote_average, poster_path, genre_ids, actorClass, id } = movieData;
   const { genresArray } = useAppSelector((state) => state.genres);
+  const dispatch = useAppDispatch();
   const containerClasses = classNames(style.container, { [style.actor__class]: actorClass });
   const imgClasses = classNames(style.img, { [style.actor__img]: actorClass });
 
+  useEffect(() => {
```

Comment:
If the condition genresArray.length === 0 will equal true, this useEffect will be called in every card. I think this is not the best place to do such call to API.